login/profile_user_blackbox_test.go

```diff
@@ -0,0 +1,152 @@
+package login_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/fabric8-services/fabric8-auth/configuration"
+	"github.com/fabric8-services/fabric8-auth/login"
+	"github.com/fabric8-services/fabric8-auth/resource"
+	testsuite "github.com/fabric8-services/fabric8-auth/test/suite"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-auth/auth"
+	keycloakLink "github.com/fabric8-services/fabric8-auth/login/link"
+
+	"github.com/goadesign/goa"
+	_ "github.com/lib/pq"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ProfileUserBlackBoxTest struct {
+	testsuite.RemoteTestSuite
+	clean                func()
+	profileService       login.UserProfileService
+	loginService         *login.KeycloakOAuthProvider
+	idpLinkService       keycloakLink.KeycloakIDPService
+	protectedAccessToken string
+	userAPIFOrAdminURL   string
+	tokenEndpoint        string
+}
+
+func TestRunProfileUserBlackBoxTest(t *testing.T) {
+	suite.Run(t, &ProfileUserBlackBoxTest{RemoteTestSuite: testsuite.NewRemoteTestSuite()})
+}
+
+// SetupSuite overrides the RemoteTestSuite's function but calls it before doing anything else
+// The SetupSuite method will run before the tests in the suite are run.
+func (s *ProfileUserBlackBoxTest) SetupSuite() {
+	resource.Require(s.T(), resource.Remote)
+	var err error
+	s.Config, err = configuration.GetConfigurationData()
+	if err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+
+	keycloakUserProfileService := login.NewKeycloakUserProfileClient()
+	s.profileService = keycloakUserProfileService
+
+	s.idpLinkService = keycloakLink.NewKeycloakIDPServiceClient()
+
+	r := &goa.RequestData{
+		Request: &http.Request{Host: "api.example.org"},
+	}
+
+	s.tokenEndpoint, err = s.Config.GetKeycloakEndpointToken(r)
+
+	// http://sso.prod-preview.openshift.io/auth/admin/realms/fabric8/users"
+	s.userAPIFOrAdminURL, err = s.Config.GetKeycloakEndpointUsers(r)
+
+	token, err := s.generateProtectedAccessToken()
```

Comment:
While you are skipping the test above the SetupSuite is not skipped and you call KC to generate a PAT. Let's move this code into the test method so we can skill all interactions with KC in the test suite.